vsintegration/src/FSharp.Editor/Common/Pervasive.fs

```diff
@@ -199,6 +223,41 @@ module Async =
             }
         async { return! agent.PostAndAsyncReply id }
 
+
+type Async with
+    /// Better implementation of Async.AwaitTask that correctly passes the exception of a failed task to the async mechanism
+    static member AwaitTaskCorrect (task:Task) : unit Async =
+        Async.FromContinuations (fun (successCont,exceptionCont,_cancelCont) ->
```

Comment:
If this ``AwaitTask`` is better, then we should start an RFC to provide a version of it in FSharp.Core please?